subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoAggregatedReport.java

```diff
@@ -0,0 +1,166 @@
+
+            executionData(getTestCategories().map(categories -> {
+                ConfigurableFileCollection coverageFiles = project.files();
+                for (String testCategory : categories) {
+                    TaskContainer tasks = project.getTasks();
+                    if (tasks.findByName(testCategory) != null) {
+                        coverageFiles.from(project.getTasks().named(testCategory).map(task -> task.getExtensions().getByType(JacocoTaskExtension.class).getDestinationFile()));
+                    }
+                }
+                return coverageFiles;
+            }));
+        });
+
+        Configuration aggregationConfiguration = project.getConfigurations().findByName(AGGREGATION_CONFIGURATION_NAME);
+        resolveClassesVariantsFrom(aggregationConfiguration, jvmEcosystemUtilities);
+        resolveSourcesVariantsFrom(aggregationConfiguration);
+        resolveTestCoverageDataVariantsFrom(aggregationConfiguration);
+    }
+
+    private void resolveClassesVariantsFrom(Configuration aggregationConfiguration, JvmEcosystemUtilities jvmEcosystemUtilities) {
+        Configuration coverageClassesDirs = createResolver("coverageClassesDirs");
+        coverageClassesDirs.extendsFrom(aggregationConfiguration);
+        coverageClassesDirs.getResolutionStrategy().getComponentSelection().all(s -> s.reject("external dependencies are excluded from code coverage report"));
+        jvmEcosystemUtilities.configureAsRuntimeClasspath(coverageClassesDirs);
+        additionalClassDirs(coverageClassesDirs.getIncoming().artifactView(it -> it.lenient(true)).getFiles());
+    }
+
+    private void resolveSourcesVariantsFrom(Configuration aggregationConfiguration) {
+        Configuration sourcesPath = createResolver("sourcesPath");
+        sourcesPath.extendsFrom(aggregationConfiguration);
+        sourcesPath.attributes(sourceDirectoriesAttributes(getProject()));
+        additionalSourceDirs(sourcesPath.getIncoming().artifactView(it -> it.lenient(true)).getFiles());
+    }
+
+    private void resolveTestCoverageDataVariantsFrom(Configuration aggregationConfiguration) {
+        Project project = getProject();
+        executionData(getTestCategories().map(categories -> {
+            ConfigurableFileCollection coverageFiles = project.files();
+            for (String testCategory : categories) {
+                String resolverName = testCategory + "CoverageDataPath";
+                Configuration coverageDataPath = project.getConfigurations().findByName(resolverName);
+                if (coverageDataPath == null) {
+                    coverageDataPath = createResolver(resolverName);
+                    coverageDataPath.extendsFrom(aggregationConfiguration);
+                    coverageDataPath.attributes(coverageDataAttributes(project, testCategory));
+                }
+                coverageFiles.from(coverageDataPath.getIncoming().artifactView(it -> it.lenient(true)).getFiles());
```

Comment:
Just checking that using an `ArtifactView` is the trick for ignoring projects that do not have `jacoco` applied but appear in the transitive set, correct?

Reply:
This as well as any external dependencies that will also not have the jacoco variants.